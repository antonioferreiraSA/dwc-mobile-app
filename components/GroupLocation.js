import React from 'react';
import { StyleSheet, View } from 'react-native';
import { Text, Heading } from './shared/Typography';

export default ({ isOnline = false, city }) => {
  if (isOnline) {
    return (
      <View style={styles.container}>
        <Heading>Location</Heading>
        <Text style={styles.line}>Meets Online</Text>
      </View>
    );
  }

  if (city) {
    return (
      <View style={styles.container}>
        <Heading>Location</Heading>
        <Text>{city}</Text>
      </View>
    );
  }

  return null;
};

const styles = StyleSheet.create({
  container: { marginBottom: 16 },
  line: { marginBottom: 10 },
});
