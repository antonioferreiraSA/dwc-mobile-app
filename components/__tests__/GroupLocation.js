import React from 'react';
import { render } from '@testing-library/react-native';
import GroupLocation from '../GroupLocation';

describe('<GroupLocation/>', () => {
  test('render a group with a location', async () => {
    const { toJSON } = render(
      <GroupLocation isOnline={false} city="San Jose" />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  test('render a group with a location, but no description', async () => {
    const { toJSON } = render(
      <GroupLocation isOnline={false} city="Sunnyvale" />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  test('render an online group', async () => {
    const { toJSON } = render(<GroupLocation isOnline />);

    expect(toJSON()).toMatchSnapshot();
  });
});
