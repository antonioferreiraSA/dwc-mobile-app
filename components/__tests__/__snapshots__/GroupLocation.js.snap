// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupLocation/> render a group with a location 1`] = `
<View
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <Text
    bold={true}
    selectable={true}
    style={
      Array [
        Object {
          "color": "#cbd2d6",
          "fontFamily": "NunitoSans-Bold",
          "fontSize": 18,
          "lineHeight": 24,
          "textAlign": "auto",
        },
        Array [
          Object {
            "color": "#cbd2d6",
            "fontSize": 18,
            "lineHeight": 24,
          },
          undefined,
        ],
      ]
    }
  >
    Location
  </Text>
  <Text
    selectable={true}
    style={
      Array [
        Object {
          "color": "#cbd2d6",
          "fontFamily": "NunitoSans-Regular",
          "fontSize": 18,
          "lineHeight": 24,
          "textAlign": "auto",
        },
        undefined,
      ]
    }
  >
    San Jose
  </Text>
</View>
`;

exports[`<GroupLocation/> render a group with a location, but no description 1`] = `
<View
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <Text
    bold={true}
    selectable={true}
    style={
      Array [
        Object {
          "color": "#cbd2d6",
          "fontFamily": "NunitoSans-Bold",
          "fontSize": 18,
          "lineHeight": 24,
          "textAlign": "auto",
        },
        Array [
          Object {
            "color": "#cbd2d6",
            "fontSize": 18,
            "lineHeight": 24,
          },
          undefined,
        ],
      ]
    }
  >
    Location
  </Text>
  <Text
    selectable={true}
    style={
      Array [
        Object {
          "color": "#cbd2d6",
          "fontFamily": "NunitoSans-Regular",
          "fontSize": 18,
          "lineHeight": 24,
          "textAlign": "auto",
        },
        undefined,
      ]
    }
  >
    Sunnyvale
  </Text>
</View>
`;

exports[`<GroupLocation/> render an online group 1`] = `
<View
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <Text
    bold={true}
    selectable={true}
    style={
      Array [
        Object {
          "color": "#cbd2d6",
          "fontFamily": "NunitoSans-Bold",
          "fontSize": 18,
          "lineHeight": 24,
          "textAlign": "auto",
        },
        Array [
          Object {
            "color": "#cbd2d6",
            "fontSize": 18,
            "lineHeight": 24,
          },
          undefined,
        ],
      ]
    }
  >
    Location
  </Text>
  <Text
    selectable={true}
    style={
      Array [
        Object {
          "color": "#cbd2d6",
          "fontFamily": "NunitoSans-Regular",
          "fontSize": 18,
          "lineHeight": 24,
          "textAlign": "auto",
        },
        Object {
          "marginBottom": 10,
        },
      ]
    }
  >
    Meets Online
  </Text>
</View>
`;
